File: smartPlantSpringBoot/_/src/main/java/com/app/tddt4iots/entities/Plant.java
package com.app.tddt4iots.entities;

import com.app.tddt4iots.enums.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.*;
import java.lang.*;

import jakarta.persistence.*;

@Entity
@Table(name = "Plant")
@Data
@NoArgsConstructor
public class Plant {
    
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
     
    @Column(name = "commonName", nullable = true, unique = true, length = 30) 
    private String commonName; 
 
    @Column(name = "scientificName", nullable = false, unique = false, length = 30) 
    private String scientificName; 
 
    @Column(name = "image", nullable = false, unique = false) 
    private Array<String> image; 
 
    @Column(name = "family", nullable = false, unique = false, length = 30) 
    private String family; 
 
    @Column(name = "kingdom", nullable = false, unique = false, length = 30) 
    private String kingdom; 
 
    @JoinColumn(name = "idSmartPot", referencedColumnName = "id") 
    @OneToOne 
    private SmartPot smartPot; 
 
    
}    

File: smartPlantAngular/src/app/Models/Plant.ts
import {SmartPot} from "./SmartPot";
export class Plant {
  get smartPot(): SmartPot {
    return this._smartPot;
  }

  set smartPot(value: SmartPot) {
    this._smartPot = value;
  }
  get kingdom(): string {
    return this._kingdom;
  }

  set kingdom(value: string) {
    this._kingdom = value;
  }
  get family(): string {
    return this._family;
  }

  set family(value: string) {
    this._family = value;
  }
  get image(): string[] {
    return this._image;
  }

  set image(value: string[]) {
    this._image = value;
  }
  get scientificName(): string {
    return this._scientificName;
  }

  set scientificName(value: string) {
    this._scientificName = value;
  }
  get commonName(): string {
    return this._commonName;
  }

  set commonName(value: string) {
    this._commonName = value;
  }
  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  private _id!: number;
  private _commonName!: string;
  private _scientificName!: string;
  private _image!: string[];
  private _family!: string;
  private _kingdom!: string;
  private _smartPot!: SmartPot;
}
